Source/Scene/Cesium3DTileBatchTable.js

```diff
@@ -1027,11 +1059,10 @@ define([
                 '    tile_diffuse = tile_diffuse_final(source, tile_featureColor); \n' +
                 '    tile_main(); \n';
         } else if (type === 'sampler2D') {
-            // Regex handles up to one level of nested parentheses:
+            // Handles any number of nested parentheses
             // E.g. texture2D(u_diffuse, uv)
             // E.g. texture2D(u_diffuse, computeUV(index))
-            regex = new RegExp('texture2D\\(' + diffuseAttributeOrUniformName + '.*?(\\)\\)|\\))', 'g');
-            source = source.replace(regex, 'tile_diffuse_final($&, tile_diffuse)');
+            source = replaceDiffuseTextureCalls(source, diffuseAttributeOrUniformName);
```

Comment:
The regex was too fragile before. It failed for a GLSL line from `processPbrMetallicRoughness`. Updated to support any number of nested parentheses.

Reply:
I didn't see a spec update for this. Can we add a test with the offending line since it sounds like a good test case?

Reply:
This code is tested in `Cesium3DTilesetSpec` with the `colorBlendMode` tests. I checked that this area of the code has 100% code coverage (the file is 99%).

Reply:
`sets colorBlendMode for textured tileset` passes in master, fails without the `replaceDiffuseTextureCalls` fix, and passes with the `replaceDiffuseTextureCalls` fix. So no update is required.